fmn/api/messaging.py

```diff
@@ -30,6 +30,14 @@ def _publish(message):
     api.publish(message)
 
 
+_background_tasks = set()
+
+
 async def publish(message):
     # Fire and forget
-    asyncio.ensure_future(run_in_threadpool(_publish, message=message))
+    # https://docs.python.org/3/library/asyncio-task.html#asyncio.create_task
+    task = asyncio.create_task(run_in_threadpool(_publish, message=message))
+    _background_tasks.add(task)
+    # To prevent keeping references to finished tasks forever, make each task
+    # remove its own reference from the set after completion:
+    task.add_done_callback(_background_tasks.discard)
```

Comment:
It actually is, it's an issue discovered by the "bugbear" plugin (or rather its reimplementation in ruff). As stated in `create_task()`'s doc linked above, it's necessary to hold a reference to future/tasks created by `ensure_future` and `create_task`, otherwise the task may be garbage-collected before it has had a chance to run. They recommend the trick of adding the reference to a list and removing it on completion.

Reply:
👍